### DISCUSSION TITLE
Wine compatibility

### DISCUSSION THREAD

**Discussion by ansani** [ansani/Shareaza#28] [General]
Do you know that Shareaza can run on Wine (both macOS and Linux)? This is still one of the goals of my release.
Please feel free to add a comment / open an issue when something goes wrong with it.

**Comment by cardpuncher** [comment]
Hi there,

So glad someone is picking up the development of Shareaza. I gave it a try on Ubuntu 22.04 with wine-7.18 (Staging) (not from Ubuntu repositories) and there are problems: after installing and completing the setup the program just disappears. When I try to relaunch it with wine Shareaza.exe, I get the following on the terminal:

`$ wine Shareaza.exe 
ERROR: ld.so: object 'libgtk3-nocsd.so.0' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
ERROR: ld.so: object 'libgtk3-nocsd.so.0' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
ERROR: ld.so: object 'libgtk3-nocsd.so.0' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
ERROR: ld.so: object 'libgtk3-nocsd.so.0' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
0698:fixme:nls:RtlGetThreadPreferredUILanguages 00000038, 00A5F6D8, 00A5F6E8 00A5F6DC
0698:fixme:nls:get_dummy_preferred_ui_language (0x38 00A5F6D8 00A5F6E8 00A5F6DC) returning a dummy value (current locale)
0698:fixme:nls:RtlGetThreadPreferredUILanguages 00000038, 00A5FA74, 00A5FA84 00A5FA78
0698:fixme:nls:get_dummy_preferred_ui_language (0x38 00A5FA74 00A5FA84 00A5FA78) returning a dummy value (current locale)
0698:fixme:ole:CoInitializeSecurity 00000000, -1, 00000000, 00000000, 4, 3, 00000000, 0, 00000000 stub
0698:fixme:msg:ChangeWindowMessageFilter 3e0 00000001
0698:fixme:msg:ChangeWindowMessageFilter 233 00000001
0698:fixme:msg:ChangeWindowMessageFilter 49 00000001
0698:fixme:msg:ChangeWindowMessageFilter 4a 00000001
0698:fixme:msg:ChangeWindowMessageFilter 111 00000001
0698:fixme:msg:ChangeWindowMessageFilter 10 00000001
0698:fixme:shcore:SetCurrentProcessExplicitAppUserModelID L"Shareaza": stub`
This is with iShareaza_2.7.10.4_Win32_Release_2022-10-12.exe.

In the next few days I hope to be able to test on my Fedora box too, so if you want me to test something specific feel free to yell at me and keep in mind I may need some hand holding.

**Reply by ansani** [reply]
Hi!
The errors you reported are related to a missing gtk .so library. Try to install it using the apt package manager.

**Reply by cardpuncher** [reply]
Hi,
It's an Ubuntu bug, the libgtk3-nocsd0 package is already installed but it makes no difference (regarding the "ERROR: ld.so: object 'libgtk3-nocsd.so.0' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored." messages). Anyway, this shouldn't be related to the "crash" of Shareaza as it's only about turning off client side decorations and these messages occur with every single program launched with wine and they mostly work fine. My guess is that the following fixme lines may give a better idea of what's going on.

Also, if using a Debug version would be more useful in providing well, debug data, I can do that. I just don't know much about the difference between the release and debug versions.

**Reply by ansani** [reply]
Difference between Release and Debug package is related to presence of debug symbols and lack of code optimizations during the build process.

I suggest to use the Release package to check if at least the app is running. I think the crash is something related to a missing library. Can you attach all the output log of Wine (if is longer than the one you already pasted).

I will try with my Ubuntu + Wine environment and check the status.

S

**Reply by cardpuncher** [reply]
I finally tried with the latest release package (v2.7.10.4-beta4) on wine-7.21 (Staging) and I couldn't make it crash. It seems to work pretty well, the only problem I encountered is that some windows are too small to display their contents, especially in the Settings window. See the screenshots below:

![Screenshot at 2022-11-18 16-33-18](https://user-images.githubusercontent.com/64688741/202719251-3feeafa4-c726-4bbe-a688-a198b3dda58b.gif)
![Screenshot at 2022-11-18 16-33-44](https://user-images.githubusercontent.com/64688741/202719319-04b2d2cc-5d31-4209-97fe-a199ca73e3d3.gif)

**Comment by ansani** [comment]
This issue is already fixed in an intermediate prerelease (https://github.com/ansani/Shareaza/actions/runs/3406005560#artifacts).
The new release will also include these fixes.

Thank you for your feedback!
